src/software/ai/navigator/path_planner/enlsvg_path_planner.cpp

```diff
@@ -0,0 +1,199 @@
+#include "enlsvg_path_planner.h"
+
+EnlsvgPathPlanner::EnlsvgPathPlanner(const Rectangle &navigable_area,
+                                     const std::vector<ObstaclePtr> &obstacles,
```

Comment:
would it make sense to construct this with a motion constraint set instead or add a constructor that does that?

Reply:
I was thinking that this `EnlsvgPathPlanner` will just be a regular path planner, and `GlobalPathPlanner` would do the heavy lifting of obstacle creation and everything using the motion constrains